cpp/log/cert.cc

```diff
@@ -710,40 +714,25 @@ StatusOr<string> Cert::SPKI() const {
     // is bad until proven otherwise.
     LOG(WARNING) << "Failed to serialize the Subject Public Key Info";
     LOG_OPENSSL_ERRORS(WARNING);
-    return util::Status(Code::INVALID_ARGUMENT, "SPKI() failed");
+    return util::Status(Code::INVALID_ARGUMENT, "Cert::SPKI() failed");
   }
 
   string result;
-  result.assign(string(reinterpret_cast<char*>(der_buf), der_length));
+  result.assign(
+      string(reinterpret_cast<char*>(CHECK_NOTNULL(der_buf)), der_length));
 
   OPENSSL_free(der_buf);
   return result;
 }
 
 
 util::Status Cert::SPKISha256Digest(string* result) const {
-  if (!IsLoaded()) {
-    LOG(ERROR) << "Cert not loaded";
-    return util::Status(Code::FAILED_PRECONDITION, "Cert not loaded");
-  }
-
-  unsigned char* der_buf(nullptr);
-  int der_length =
-      i2d_X509_PUBKEY(X509_get_X509_PUBKEY(x509_.get()), &der_buf);
-  if (der_length < 0) {
-    // Failed to serialize. Several possible reasons but we will just reject
-    // the input rather than trying to interpret the cause
-    LOG(WARNING) << "Failed to serialize the Subject Public Key Info";
-    LOG_OPENSSL_ERRORS(WARNING);
-    return util::Status(Code::INVALID_ARGUMENT, "SPKI SHA256 digest failed");
+  const util::StatusOr<string> spki(SPKI());
+  if (spki.ok()) {
+    string sha256_digest = Sha256Hasher::Sha256Digest(spki.ValueOrDie());
+    result->assign(sha256_digest);
```

Comment:
Why don't people just do `*result = Sha256Hasher::Sha256Digest(spki.ValueOrDie());`? Or, if you like the temporary (to give it a name, as a form of documentation?), `*result = sha256_digest`?

Not your doing, but could you at least add a `CHECK_NOTNULL`, while you're at it?